website/project/model.py

```diff
@@ -75,16 +74,9 @@ def has_anonymous_link(node, auth):
     :param str link: any view-only link in the current url
     :return bool anonymous: Whether the node is anonymous to the user or not
     """
-    view_only_link = auth.private_key or request.args.get('view_only', '').strip('/')
```

Comment:
`Auth` is instantiated in a lot of places. That strikes me as an anti-pattern (though, again, out of scope for this PR).

```
website/addons/base/testing.py:72:        self.project.add_addon(self.ADDON_SHORT_NAME, auth=Auth(self.user))
website/addons/base/views.py:251:        auth = Auth(user=user)
website/addons/box/model.py:151:        nodelogger = BoxNodeLogger(node=self.owner, auth=Auth(user_settings.owner))
website/addons/box/model.py:202:        self.deauthorize(Auth(user=user), add_log=True)
website/addons/dataverse/model.py:269:        self.deauthorize(Auth(user=user), add_log=True)
website/addons/dataverse/provider.py:48:            auth=Auth(user=user),
website/addons/dataverse/provider.py:59:        node_addon.deauthorize(Auth(user=user))
website/addons/dropbox/model.py:49:            node_settings.deauthorize(Auth(self.owner))
website/addons/dropbox/model.py:95:        nodelogger = DropboxNodeLogger(node=self.owner, auth=Auth(user_settings.owner))
website/addons/dropbox/model.py:255:        self.deauthorize(Auth(user=user), add_log=True)
website/addons/dropbox/templates/dropbox_user_settings.mako:9:            <span data-bind="if: userHasAuth() && loaded()">
website/addons/dropbox/templates/dropbox_user_settings.mako:16:            <span data-bind="if: !userHasAuth() && loaded()">
website/addons/figshare/model.py:41:            node_settings.deauthorize(auth=Auth(user=self.owner), save=True)
website/addons/figshare/model.py:115:            auth=Auth(user=user_settings.owner),
website/addons/figshare/model.py:369:        self.deauthorize(Auth(user=user), add_log=True, save=True)
website/addons/github/model.py:175:            auth=Auth(user_settings.owner),
website/addons/github/model.py:525:        self.deauthorize(Auth(user=user), log=True, save=True)
website/addons/googledrive/model.py:312:        self.deauthorize(Auth(user=user), add_log=True, save=True)
website/addons/osfstorage/views.py:78:    is_anon = has_anonymous_link(node_addon.owner, Auth(private_key=request.args.get('view_only')))
website/addons/osfstorage/views.py:203:    auth = Auth(User.load(payload['user']))
website/addons/s3/model.py:76:            auth=Auth(user_settings.owner),
website/addons/s3/model.py:265:        self.deauthorize(Auth(user=user), log=True, save=True)
website/archiver/utils.py:106:    addon = node.get_or_add_addon(settings.ARCHIVE_PROVIDER, auth=Auth(user))
website/conferences/utils.py:74:    auth = Auth(user=user)
website/project/decorators.py:171:    if not node.can_view(Auth(user=user)) and api_node != node:
website/project/model.py:773:                self.add_tag(tag, Auth(self.creator), save=False, log=False)
website/project/model.py:1384:                auth=Auth(user=self.creator),
website/project/model.py:3103:            auth=Auth(user),
website/project/model.py:3161:            auth=Auth(user),
website/project/model.py:3165:            self.set_privacy('private', Auth(user))
website/project/model.py:3197:            auth=Auth(user),
website/project/model.py:3691:            auth=Auth(user),
website/project/model.py:3715:            auth=Auth(self.initiated_by),
website/project/model.py:3810:            auth=Auth(user),
website/project/model.py:3822:            auth=Auth(self.initiated_by),
website/project/model.py:3833:                auth=Auth(self.initiated_by),
website/project/model.py:3837:        auth = Auth(self.initiated_by)
website/project/model.py:3925:            auth=Auth(user),
website/project/model.py:3935:        auth = Auth(self.initiated_by)
website/project/model.py:3940:        auth = Auth(user) if user else None
website/project/model.py:3965:            auth=Auth(user),
website/project/model.py:3998:        auth = Auth(draft.initiator)
api/base/serializers.py:173:        auth = auth_core.Auth(user)
api/base/utils.py:40:        auth = Auth(None)
api/base/utils.py:42:        auth = Auth(user)
api/collections/views.py:248:        auth = Auth(user)
api/collections/views.py:474:        auth = Auth(user)
api/comments/serializers.py:52:        auth = Auth(user)
api/comments/serializers.py:64:        auth = Auth(self.context['request'].user)
api/nodes/serializers.py:132:            auth = Auth(None)
api/nodes/serializers.py:134:            auth = Auth(user)
api/nodes/serializers.py:254:        auth = Auth(self.context['request'].user)
api/nodes/serializers.py:277:        auth = Auth(self.context['request'].user)
api/nodes/serializers.py:329:        auth = Auth(user)
api/users/views.py:297:            auth = Auth(None)
api/users/views.py:299:            auth = Auth(current_user)
framework/auth/core.py:147:class Auth(object):
framework/auth/core.py:157:        return ('<Auth(user="{self.user}", '
framework/auth/core.py:1268:                        node.set_visible(user=self, visible=True, log=True, auth=Auth(user=self))
framework/auth/core.py:1281:                        auth=Auth(user=self),
```